reactor-client/src/main/java/com/aerospike/client/reactor/AerospikeReactorClient.java

```diff
@@ -0,0 +1,467 @@
+						}
+					}
+					@Override
+					public void onFailure(AerospikeException exception) {
+						sink.error(exception);
+					}
+				},
+				policy, key));
+	}
+
+	@Override
+	public final Mono<KeysExists> exists(Key[] keys) throws AerospikeException {
+		return exists(null, keys);
+	}
+
+	@Override
+	public final Mono<KeysExists> exists(BatchPolicy policy, Key[] keys) throws AerospikeException{
+		return Mono.create(sink -> aerospikeClient.exists(eventLoops.next(), new ExistsArrayListener() {
+					@Override
+					public void onSuccess(Key[] keys, boolean[] exists) {
+						sink.success(new KeysExists(keys, exists));
+					}
+
+					@Override
+					public void onFailure(AerospikeException exception) {
+						sink.error(exception);
+					}
+				},
+				policy, keys));
+	}
+
+	@Override
+	public final Flux<KeyExists> existsFlux(Key[] keys) throws AerospikeException {
+		return existsFlux(null, keys);
+	}
+
+	@Override
+	public final Flux<KeyExists> existsFlux(BatchPolicy policy, Key[] keys) throws AerospikeException {
+		return Flux.create(sink -> aerospikeClient.exists(eventLoops.next(), new ExistsSequenceListener() {
+					@Override
+					public void onExists(Key key, boolean exists) {
+						sink.next(new KeyExists(key, exists));
+					}
+					@Override
+					public void onSuccess() {
+						sink.complete();
+					}
+					@Override
+					public void onFailure(AerospikeException exception) {
+						sink.error(exception);
+					}
+				},
+				policy, keys));
+	}
+
+	@Override
+	public final Mono<Key> put(Key key, Bin... bins) throws AerospikeException {
+		return put(null, key, bins);
+	}
+
+	@Override
+	public final Mono<Key> put(WritePolicy policy, Key key, Bin... bins) throws AerospikeException {
+		return Mono.create(sink -> aerospikeClient.put(eventLoops.next(), new WriteListener() {
+					@Override
+					public void onSuccess(Key key) {
+						sink.success(key);
+					}
+					@Override
+					public void onFailure(AerospikeException exception) {
+						sink.error(exception);
+					}
+				},
+				policy, key, bins));
+	}
+
+	@Override
+	public final Mono<Key> append(Key key, Bin... bins) throws AerospikeException {
+		return append(null, key, bins);
+	}
+
+	@Override
+	public final Mono<Key> append(WritePolicy policy, Key key, Bin... bins) throws AerospikeException {
+		return Mono.create(sink -> aerospikeClient.append(eventLoops.next(), new WriteListener() {
+					@Override
+					public void onSuccess(Key key) {
+						sink.success(key);
+					}
+					@Override
+					public void onFailure(AerospikeException exception) {
+						sink.error(exception);
+					}
+				},
+				policy, key, bins));
+	}
+
+	@Override
+	public final Mono<Key> prepend(Key key, Bin... bins) throws AerospikeException {
+		return prepend(null, key, bins);
+	}
+
+	@Override
+	public final Mono<Key> prepend(WritePolicy policy, Key key, Bin... bins) throws AerospikeException {
+		return Mono.create(sink -> aerospikeClient.prepend(eventLoops.next(), new WriteListener() {
```

Comment:
Although AsyncClient is deprecated retrieving event loop has some additional checks. Could you please check `AsyncClient.findEventLoop`?